File: Leetcode_55.java
class Solution {
    public boolean canJump(int[] nums) {
        int reachable = 0;  // Maximum index we can currently reach
        
        for (int i = 0; i < nums.length; i++) {
            if (i > reachable) {
                return false;  // If we can't reach this index, return false
            }
            reachable = Math.max(reachable, i + nums[i]);  // Update the farthest we can go
        }
        
        return true;  // If loop completes, last index is reachable
    }
}

File: Leetcode_55.py
class Solution:
    def canJump(self, nums: List[int]) -> bool:
        reachable = 0
        for i in range(len(nums)):
            if i > reachable:
                return False
            reachable = max(reachable, i + nums[i])
        return True


'''Another Sol
class Solution:
    def canJump(self, nums: List[int]) -> bool:
        i = 0
        while i < len(nums):
            if i + nums[i] >= len(nums) - 1:
                return True
            if nums[i] == 0:
                return False
            i += 1
        return False
'''
